src/utils/validations/component/validations/with-validation.spec.js

```diff
@@ -0,0 +1,383 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import PropTypes from 'prop-types';
+import { withValidation } from '.';
+import guid from '../../../helpers/guid/guid';
+import Icon from '../../../../components/icon';
+
+const presErr = new Error('this value is required!');
+
+const notZeroErr = new Error('this is zero!');
+const asyncErr = new Error('not foo!');
+const failErr = new Error("It's foo!");
+
+const presence = value => new Promise((resolve, reject) => {
+  if (value) {
+    resolve(true);
+  } else {
+    reject(presErr);
+  }
+});
+
+const isNotZero = value => new Promise((resolve, reject) => {
+  if (value !== 1) {
+    resolve(true);
+  } else {
+    reject(notZeroErr);
+  }
+});
+
+const asyncValidation = value => new Promise((resolve, reject) => {
+  const bound = Math.floor(Math.random() * 251);
+  setTimeout(() => {
+    if (value === 'foo') {
+      resolve();
+    } else {
+      reject(asyncErr);
+    }
+  }, bound);
+});
+
+const failValidation = val => new Promise((resolve, reject) => {
+  if (val !== 'foo') resolve(true);
+  reject(failErr);
+});
+
+const context = { addInput: () => 'foo', removeInput: () => 'foo', adjustCount: () => 'foo' };
+
+const MockComponent = props => <input { ...props } />;
+
+const InputComponent = withValidation(MockComponent);
+
+InputComponent.contextTypes = {
+  addInput: PropTypes.func,
+  removeInput: PropTypes.func,
+  adjustCount: PropTypes.func
+};
+
+const WithNoDisplayName = withValidation(() => <input />);
+
+const types = ['error', 'warning', 'info'];
+
+const shallowRenderWithContext = props => shallow(
+  <InputComponent name={ guid() } { ...props } />, { context }
+);
+
+const validations = [isNotZero, presence, asyncValidation];
+
+describe('when the withValidations HOC wraps a component', () => {
+  let wrapper, wrapper2;
+
+  beforeEach(() => {
+    wrapper = shallow(<InputComponent name='foo' />, { context });
+  });
+
+  it('matches the snapshot with the input component wrapped in the HOC with default props', () => {
+    wrapper2 = shallowRenderWithContext({ name: 'foo ' });
+    expect(wrapper2).toMatchSnapshot();
+  });
+
+  it('matches the snapshot when the HOC is passed an anonymous component with no name or display name', () => {
+    wrapper2 = shallow(<WithNoDisplayName name='foo' />);
+    expect(wrapper2).toMatchSnapshot();
+  });
+
+  it('calls the onBlur function to handle the blur event and start the validations of the component', () => {
+    const spy = jasmine.createSpy('blurred');
+    wrapper.instance().handleBlur();
+    expect(spy).not.toHaveBeenCalled();
```

Comment:
Seems unnecessary to test whether the `spy` is called _before_ you attach the `spy`?

Reply:
Yeah no idea what my intention was there haha, removed